### PR TITLE
OCPBUGS-9920 Issue in file networking/metallb/metallb-upgrading-operator.adoc

### PR DESCRIPTION
**Pull request by kquinn1204**
[OCPBUGS-9920]: Upgrade does not address upgrade to latest version has hardcoded 4.11

Version(s):
main, 4.12 and 4.13
Issue:
https://issues.redhat.com/browse/OCPBUGS-9920

Link to docs preview:
https://57583--docspreview.netlify.app/openshift-enterprise/latest/networking/metallb/metallb-upgrading-operator.html

QE review:
- [x] QE has approved this change.

Additional information:
Will overide these changes for 4.12 https://github.com/openshift/openshift-docs/pull/56265 but with this change it should be more future proof and we will not have to keep updating with newer releases like 4.13, 4.14 etc..

### CODE REVIEW

**Comment by kquinn1204** [comment]
@fedepaol @asood-rh I updated the upgrade section to ensure it accurately reflects for all version prior to 4.10 the upgrade to the latest version whatever that may be. Please check this PR and if it is still valid for 4.12 4.13 etc.

**Comment by asood-rh** [comment]
@kquinn1204 Yes, Upgrade from 4.10 -> 4.11 or 4.12 or 4.13 will need 4.10 is deleted.
Upgrade from 4.11->4.12->4.13 should be seamless.
Another PR https://github.com/openshift/openshift-docs/pull/56265 for 4.12

**Comment by kquinn1204** [comment]
Thank you I will look at integrating that PR here and then get this merged. @asood-rh do you think that is a good way forward?

**Comment by asood-rh** [comment]
Hi @kquinn1204 It will be the case till 4.10 is EOL. I believe issue is scoping of operator in 4.10. Post 4.10 operator is Allnamespaces scoped.

**Comment by kquinn1204** [comment]
Hi @asood-rh can you review this and give an LGTM I am hoping my clarification will ensure this guidance can carry forward until as you say 4.10 is end of life.

**Comment by asood-rh** [comment]
@kquinn1204 **4.10 is not EOL yet**, it will eventually when we have more releases of openshift, that is when we can stop adding the comment.

**Comment by kquinn1204** [comment]
Thanks for reviewing

TL;DR No @eohartman only versions called out. I believe it does not as the issue with upgrade only was recognized in 4.11 so the statement in 4.11 is valid for 4.11. However, this statement "Upgrading from any version greater than OpenShift Container Platform 4.10 to the latest version works as expected." should future-proof it for future version. If I read 4.12 docs that will be the latest version with the previous version being 4.11 (any version greater than OpenShift Container Platform 4.10). I am trying to avoid harcoding versions in that statement and then having to revise for 4.13, 4.14 etc.

File: networking/metallb/metallb-upgrading-operator.adoc [hunk 1]
```diff
@@ -6,18 +6,18 @@ include::_attributes/common-attributes.adoc[]
 
 toc::[]
 
-The automatic upgrade procedure does not work as expected from {product-title} 4.10 and earlier. A summary of the upgrade procedure is as follows:
+The automatic upgrade procedure does not work as expected from {product-title} 4.10 and earlier. Upgrading from any version greater than {product-title} 4.10 to the latest version works as expected. A summary of the upgrade procedure from {product-title} 4.10 and earlier is as follows:
```
[the remark this answers is not in the document]
Comment (michaelryanpeter) [reply networking/metallb/metallb-upgrading-operator.adoc:9]: PTAL the IBM Style Guide's section on [Product versions](https://www.ibm.com/docs/en/ibm-style?topic=references-product-versions). I think there are some useful patterns that would help make this clearer. Note, use _later_ to refer to versions instead of _greater than_.

File: networking/metallb/metallb-upgrading-operator.adoc [hunk 2]
```diff
@@ -6,18 +6,18 @@ include::_attributes/common-attributes.adoc[]
...
 
-. Delete the previously installed Operator version for example 4.10. Ensure that the namespace and the `metallb` custom resource are not removed.
+. Delete the 4.10 installed MetalLB Operator version. Ensure that the namespace and the `metallb` custom resource are not removed.
 
-. Install the 4.11 version of the Operator using the CLI. Install the 4.11 version of the Operator in the same namespace that the previously installed Operator version was installed to.
+. Install the {product-version} version of the Operator using the CLI. Install the {product-version} version of the Operator in the same namespace that the previously installed Operator version was installed to.
```
[the remark this answers is not in the document]
Comment (michaelryanpeter) [reply networking/metallb/metallb-upgrading-operator.adoc:13]: If you are going to use want to use the version number to refer to the product, ISG guidance says to use _version <version_number>_ instead of _<version_number> version_.

**Review by michaelryanpeter — COMMENTED** [review]
I tried to stay in scope with my comments, but there are some minor clarity and style issues that need to be addressed in the changed portions that should be updated elsewhere for consistency. 

PTAL at the ISG's guidance on product versions: https://www.ibm.com/docs/en/ibm-style?topic=references-product-versions.

I think if you clarify a couple of small points and update your usage to the ISG, this will read very clearly. I hope this helps, and if any of my comments are unclear, please feel free to reach out for clarification. Thanks!

/remove-label merge-review-in-progress
/remove-label merge-review-needed

File: networking/metallb/metallb-upgrading-operator.adoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (kquinn1204) [reply networking/metallb/metallb-upgrading-operator.adoc:13]: for MetalLB it is tied to OCP version so when OCP rev'd it is rev'd

**Comment by kquinn1204** [comment]
@michaelryanpeter I have made some edits based on your input - thanks. Hopefully the guidance is a little clearer now and this can be merged.

File: networking/metallb/metallb-upgrading-operator.adoc [hunk 3]
```diff
@@ -1,31 +1,33 @@
...
 [id="metallb-upgrading-operator"]
-= Upgrading the MetalLB Operator
+= Upgrading the MetalLB
 include::_attributes/common-attributes.adoc[]
 :context: metallb-upgrading-operator
 
 toc::[]
 
-The automatic upgrade procedure does not work as expected from {product-title} 4.10 and earlier. A summary of the upgrade procedure is as follows:
+There is a known issue with upgrading from any version of the MetalLB Operator that is version 4.10 or earlier, where automatic updates do not work. However, upgrading from any version of the MetalLB Operator that is version 4.11 or later to a newer version is successful. For example, upgrading from version 4.12 to version 4.13 will occur smoothly.
```
Comment (mburke5678) [inline networking/metallb/metallb-upgrading-operator.adoc:9]: I agree with @michaelryanpeter that this line doesn't sound quite right. 
Maybe:
> There is a known issue with upgrading from MetalLB Operator 4.10 or earlier

File: networking/metallb/metallb-upgrading-operator.adoc [hunk 4]
```diff
@@ -1,31 +1,33 @@
...
 
-. Delete the previously installed Operator version for example 4.10. Ensure that the namespace and the `metallb` custom resource are not removed.
+A summary of the upgrade procedure for the MetalLB Operator from 4.10 and earlier is as follows:
 
-. Install the 4.11 version of the Operator using the CLI. Install the 4.11 version of the Operator in the same namespace that the previously installed Operator version was installed to.
+. Delete the installed MetalLB Operator version for example 4.10. Ensure that the namespace and the `metallb` custom resource are not removed.
+
+. Install using the CLI the version {product-version} of the MetalLB Operator in the same namespace where the previous version of the MetalLB Operator was installed.
```
Comment (mburke5678) [inline networking/metallb/metallb-upgrading-operator.adoc:15]:
```suggestion
. Using the CLI, install the MetalLB Operator {product-version} in the same namespace where the previous version of the MetalLB Operator was installed.
```

**Comment by mburke5678** [comment]
@kquinn1204 I had a couple of small nits also.

File: networking/metallb/metallb-upgrading-operator.adoc (hunk 3 above)
Comment (kquinn1204) [reply networking/metallb/metallb-upgrading-operator.adoc:9]: I cannot go with "There is a known issue in versions of the MetalLB Operator released with OCP 4.10 .." strictly speaking the Operator is not released with OCP it is available in the Operator hub OK, but I'm not sure if that means it is released with OCP 4.12. Maybe I am splitting hairs there, but that statement would suggest to me when OCP installed MetalLB is there.
